src/openforms/registrations/contrib/stuf_zds/plugin.py

```diff
@@ -219,6 +227,76 @@ def get_extra_data(
         payment_extra = _get_extra_payment_variables(submission, options)
         return {**data, **payment_extra}
 
+    def process_partners(
+        self,
+        submission: Submission,
+        zaak_data: MutableMapping[str, Any],
+        extra_data: dict[str, Any],
+    ) -> tuple[MutableMapping[str, Any], MutableMapping[str, Any]]:
+        config = StufZDSConfig.get_solo()
+        description = config.zaakbetrokkene_partners_omschrijving
+
+        # register as zaakbetrokkene
+        if RegistrationAttribute.partners in zaak_data:
+            component = get_component(
+                submission,
+                RegistrationAttribute.partners,
+                REGISTRATION_ATTRIBUTE,
+            )
+
+            assert component is not None
+            assert component["type"] == "partners"
+
+            fm_immutable_variable = (
+                FormVariable.objects.filter(
+                    source=FormVariableSources.user_defined,
+                    prefill_plugin="family_members",
+                )
+                .filter(prefill_options__mutable_data_form_variable=component["key"])
+                .first()
+            )
+
+            if fm_immutable_variable:
+                if fm_immutable_variable.key in extra_data:
+                    del extra_data[fm_immutable_variable.key]
+
+                # xml file generation depends on whether the variable was prefilled or not
+                # (authentiek)
+                state = submission.load_submission_value_variables_state()
+                submission_variable = state.get_variable(fm_immutable_variable.key)
+
+                # update the zaak data with the information needed for the xml request
+                for partner in zaak_data[RegistrationAttribute.partners]:
+                    partner.update(
+                        {
+                            **partner,
+                            "description": description,
+                            "dateOfBirth": partner["dateOfBirth"].replace("-", ""),
+                            "prefilled": bool(submission_variable),
+                        }
+                    )
+                    del partner["dateOfBirthPrecision"]
+        # register as extraElementen
+        else:
+            variables = FormVariable.objects.filter(
+                source=FormVariableSources.user_defined,
+                prefill_plugin="family_members",
+            ).exclude(prefill_options={})
+            for variable in variables:
+                from_key, to_key = (
+                    variable.key,
+                    variable.prefill_options["mutable_data_form_variable"],
+                )
+                value = extra_data[from_key]
+                for item in value:
+                    item["description"] = description
+                    del item["dateOfBirthPrecision"]
+
+                extra_data[to_key] = value
+                del extra_data[from_key]
+
+        return zaak_data, extra_data
```

Comment:
because we are doing mutation, I would not return anything from this function and update the type hint for the return value to `-> None`. That makes it crystal clear that we're mutating the dicts.